uk.co.compucorp.civicrm.hrleaveandabsences/views/my-leave/components/my-leave-calendar.html

```diff
@@ -0,0 +1,386 @@
+            <th>Fri <span>1</span></th>
+            <th class="chr_leave-calendar__weekend">Sat <span>2</span></th>
+            <th class="chr_leave-calendar__weekend">Sun <span>3</span></th>
+            <th>Mon <span>4</span></th>
+            <th>Tue <span>5</span></th>
+            <th>Wed <span>6</span></th>
+            <th>Thu <span>7</span></th>
+            <th>Fri <span>8</span></th>
+            <th class="chr_leave-calendar__weekend">Sat <span>9</span></th>
+            <th class="chr_leave-calendar__weekend">Sun <span>10</span></th>
+            <th>Mon <span>11</span></th>
+            <th>Tue <span>12</span></th>
+            <th>Wed <span>13</span></th>
+            <th>Thu <span>14</span></th>
+            <th>Fri <span>15</span></th>
+            <th class="chr_leave-calendar__weekend">Sat <span>16</span></th>
+            <th class="chr_leave-calendar__weekend">Sun <span>17</span></th>
+            <th>Mon <span>18</span></th>
+            <th>Tue <span>19</span></th>
+            <th>Wed <span>20</span></th>
+            <th>Thu <span>21</span></th>
+            <th>Fri <span>22</span></th>
+            <th class="chr_leave-calendar__weekend">Sat <span>23</span></th>
+            <th class="chr_leave-calendar__weekend">Sun <span>24</span></th>
+            <th>Mon <span>25</span></th>
+            <th>Tue <span>26</span></th>
+            <th>Wed <span>27</span></th>
+            <th>Thu <span>28</span></th>
+            <th>Fri <span>29</span></th>
+            <th class="chr_leave-calendar__weekend">Sat <span>30</span></th>
+            <th class="chr_leave-calendar__weekend">Sun <span>31</span></th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr class="chr_leave-calendar__items">
+            <td></td>
+            <td class="chr_leave-calendar__weekend"></td>
+            <td class="chr_leave-calendar__weekend"></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td>
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--type--leave"></div>
+            </td>
+            <td class="chr_leave-calendar__weekend">
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--type--leave"></div>
+            </td>
+            <td class="chr_leave-calendar__weekend">
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--type--leave"></div>
+            </td>
+            <td>
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--type--leave"></div>
+            </td>
+            <td>
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--type--leave"><span>AM</span></div>
+            </td>
+            <td></td>
+            <td></td>
+            <td>
+              <div class="chr_leave-calendar__item chr_leave-calendar__item--requested chr_leave-calendar__item--type--leave"><span>PM</span></div>
```

Comment:
For the half-days, instead of using the text `PM` or `AM` from HTML, we can create a new class and specify the text using `:after`. Then easier to build the logic from JS